### PR TITLE
refactor(stdlib): make integral return err for null or out-of-order t…

### PR DESCRIPTION
**Pull request by affo**
…imes, add support for int and uint

### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: stdlib/universe/integral.go [hunk 1]
```diff
@@ -146,37 +147,70 @@ func (t *integralTransformation) Process(id execute.DatasetID, tbl flux.Table) e
...
+		}
+
+		if tbl.Key().HasCol(c) {
+			return fmt.Errorf("cannot aggregate columns that are part of the group key")
+		}
+
+		if typ := cols[idx].Type; typ != flux.TFloat &&
+			typ != flux.TInt &&
+			typ != flux.TUInt {
+			return fmt.Errorf("cannot perform integral over %v", typ)
```
Comment (wolffcm) [inline stdlib/universe/integral.go]: Nice.  When I modified `derivative` I neglected to add checks like this, so I filed a bug for it.

Comment (wolffcm) [reply stdlib/universe/integral.go]: Apropos of nothing, it irks me that we have to do this check at run time.

File: stdlib/universe/integral.go [hunk 2]
```diff
@@ -146,37 +147,70 @@ func (t *integralTransformation) Process(id execute.DatasetID, tbl flux.Table) e
...
+				prevTime = tm
+
+				switch tbl.Cols()[j].Type {
+				case flux.TInt:
+					if vs := cr.Ints(j); vs.IsValid(i) {
+						in.updateFloat(tm, float64(vs.Value(i)))
+					}
+				case flux.TUInt:
+					if vs := cr.UInts(j); vs.IsValid(i) {
+						in.updateFloat(tm, float64(vs.Value(i)))
```
Comment (wolffcm) [inline stdlib/universe/integral.go]: Would this have panicked before for `int` and `uint` columns?   Nice catch.  I guess doing it this way we lose some numeric accuracy for 64-bit integer values that cannot be represented as a float.  (Just making the observation, not suggesting you fix this now)

Comment (affo) [reply stdlib/universe/integral.go]: Oh, good catch!  
Yep, it panicked 😅

File: stdlib/universe/integral.go (hunk 1 above)
Comment (affo) [reply stdlib/universe/integral.go]: Yeah, you are right. Anyhow, I think that being schemaless would always require runtime checks on the schema.
If we had typed streams we could do any type checking at compile time and also avoid schema collisions. However we would loose flexibility. I am always open to introduce typed streams :)

File: stdlib/universe/integral_test.go [hunk 3]
```diff
@@ -205,9 +296,58 @@ func TestIntegral_Process(t *testing.T) {
...
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+					{Label: "t", Type: flux.TString},
+				},
+				Data: [][]interface{}{
+					{execute.Time(1), execute.Time(3), execute.Time(1), 2.0, "a"},
+					{execute.Time(1), execute.Time(3), execute.Time(2), 1.0, "b"},
 				},
 			}},
+			wantErr: fmt.Errorf("cannot perform integral over string"),
```
[the remark this answers is not in the document]
Comment (affo) [reply stdlib/universe/integral_test.go]: Bug fixed, thank you :)

**Comment by affo** [comment]
Fixes #906 together with #908

**Pull request merged**